proposals/pr-contrib-guide.md

```diff
@@ -0,0 +1,219 @@
+   (especially!) if that’s saying no. An initial response to open PRs should
+   occur within [TBD] business days.
+7. Questions and friction should be reduced with automation wherever possible
+   and documented when not.
+
+### Non-goals
+
+### Motivating Examples
+
+#### Limited information on where to contribute
+
+The AWS CLI currently has code, documentation, and examples published publicly
+on a number of GitHub repositories. We have hundreds of open issues and pull
+requests across these areas. However, we do not have a guide for users to know
+what things are available for them to contribute on or what the process or
+guidelines are. We get frequent requests to contribute on specific issues, but
+it's a significant effort to respond to these requests individually.
+
+We need a well-documented guide and process for the community to know what is available to work on so that their efforts are used effectively.
+
+#### Unprioritized contributions
+
+Some pull requests have had initial contact from maintainers as desirable
+changes, but then fall off the radar and remain unprioritized. This leaves the
+impression that user requests are not a priority. For example, [`botocore` pull
+request 1231](https://github.com/boto/botocore/pull/1231) addresses releasing
+HTTP connections back to a pool after making a request. It proposes to fix an
+issue the `boto3` issue [#454](https://github.com/boto/boto3/issues/454) that
+has significant user request (60 👍🏻, 38 comments). There have been comments by
+various Python SDK team members but no resolutions proposed. The last comment
+from AWS was in 2016. Changes from previous Python SDK team members were
+requested in 2017, and then noted that offline discussion indicated bigger
+issues. The code change itself is small (< 20 lines) but is in a critical piece
+of the SDK and would have a large blast radius if changed inappropriately.
+
+We need to collect and review user feedback to decide on what to work on. We
+need to tell users how they can provide feedback in the best way for us to take
+action.
+
+#### Unsolicited pull request
+
+Many pull requests come as unsolicited changes that often do not correspond to
+an open issue that has been discussed by the maintainers or the community. They
+often end up lingering with no communication from the maintainers to indicate
+the feasibility or appropriateness of the changes. As an example, [`aws-cli`
+pull request 2636](https://github.com/aws/aws-cli/pull/2636)  adds functionality
+to configure MFA. It adds or alters 231 lines of code across five files, and is
+a significant new feature. There is no open issue or discussion about the
+feature other than on the PR. It has no comments from AWS employees, other than
+to indicate that it is a “large” pull request, since 2017. Conversation
+continued for a year on the PR by interested users with no other interaction or
+guidance from AWS.
+
+A idea that has not been vetted by the community and the maintainers is
```

Comment:
Takeaways here:
* We want to maximize efficiency. We do not want someone to send a pull request and us tell them they have to rewrite it when it would have been better to discuss it in an issue first.
* It is hard to have a discussion when it involves both code changes and proposal of the idea.
* We want to be able to prioritize against other community requests as it does take a large amount of effort to contribute and review.